gnome-initial-setup/demo/eos-demo-video.c

```diff
@@ -0,0 +1,230 @@
+      event->type == GDK_KEY_PRESS ||
+      event->type == GDK_MOTION_NOTIFY)
+    {
+      gtk_widget_destroy (widget);
+    }
+}
+
+static gboolean
+on_video_bus_event (GstBus     *bus,
+                    GstMessage *msg,
+                    gpointer   user_data)
+{
+  GisDemoVideo *demo_video = GIS_DEMO_VIDEO (user_data);
+  GisDemoVideoPrivate *priv = gis_demo_video_get_instance_private (demo_video);
+
+  switch (GST_MESSAGE_TYPE (msg))
+    {
+    case GST_MESSAGE_EOS:
+      gst_element_set_state (priv->player, GST_STATE_PAUSED);
+      gst_element_seek_simple (priv->player,
+                               GST_FORMAT_TIME,
+                               GST_SEEK_FLAG_FLUSH |
+                               GST_SEEK_FLAG_ACCURATE,
+                               0);
+      gst_element_set_state (priv->player, GST_STATE_PLAYING);
+    default:
+      break;
+    }
+
+  return G_SOURCE_CONTINUE;
+}
+
+#define DEMO_VIDEO_FILE_URI "file://" DEMOVIDEODATADIR "/eos-demo-video.ogv"
+
+/* The shell hides all windows on startup, by showing
+ * the overview which is understandable, so show the
+ * window a few seconds later */
+static gboolean
+on_show_application (gpointer user_data)
+{
+  GisDemoVideo *demo_video = GIS_DEMO_VIDEO (user_data);
+  GisDemoVideoPrivate *priv = gis_demo_video_get_instance_private (demo_video);
+
+  gtk_widget_show_all (GTK_WIDGET (priv->main_window));
+  gst_element_set_state (priv->player, GST_STATE_PLAYING);
+
+  gst_bus_add_watch (priv->bus, on_video_bus_event, demo_video);
+  return FALSE;
+}
+
+static void
+gis_demo_video_startup (GApplication *application)
+{
+  GisDemoVideo *demo_video = GIS_DEMO_VIDEO (application);
+  GisDemoVideoPrivate *priv = gis_demo_video_get_instance_private (demo_video);
+  GtkWidget *window = NULL;
+  GtkWidget *video_parent = NULL;
+
+  G_APPLICATION_CLASS (gis_demo_video_parent_class)->startup (application);
+
+  priv->main_window = GTK_WINDOW (gtk_application_window_new (GTK_APPLICATION (application)));
+  gtk_widget_set_events (GTK_WIDGET (priv->main_window),
+                         GDK_POINTER_MOTION_MASK |
+                         GDK_KEY_PRESS_MASK |
+                         GDK_BUTTON_PRESS_MASK); 
+  gtk_window_fullscreen (priv->main_window);
+  gtk_application_add_window (GTK_APPLICATION (application),
+                              priv->main_window);
+
+  /* Translators: this text appears when the user invokes
+   * some action that is blocked because the demo video
+   * is showing */
+  gtk_application_inhibit (GTK_APPLICATION (application),
+                           priv->main_window,
+                           GTK_APPLICATION_INHIBIT_LOGOUT |
+                           GTK_APPLICATION_INHIBIT_SWITCH |
+                           GTK_APPLICATION_INHIBIT_IDLE,
+                           _("Demonstration video"));
+
+  /* Reparent video into our application window */
+  video_parent = gtk_widget_get_parent (priv->widget);
+  if (video_parent)
```

Comment:
That does not make a lot of sense; the examples of gstgtk that I've seen don't do this reparenting, so I'd just remove it.